extensions/apicurio-registry-avro/runtime/src/main/java/io/quarkus/apicurio/registry/binding/ServiceRegistryBindingConverter.java

```diff
@@ -0,0 +1,105 @@
+
+    private static final String INCOMING_PREFIX = "mp.messaging.incoming.";
+    private static final String OUTGOING_PREFIX = "mp.messaging.outgoing.";
+
+    @Override
+    public Optional<ServiceBindingConfigSource> convert(List<ServiceBinding> serviceBindings) {
+        var matchingByType = ServiceBinding.singleMatchingByType("serviceregistry", serviceBindings);
+        Config config = ConfigProvider.getConfig();
+        if (matchingByType.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var binding = matchingByType.get();
+
+        List<String> channels = extractChannels(config);
+
+        Map<String, String> properties = new HashMap<>();
+
+        String registryUrl = binding.getProperties().get("registryUrl");
+        if (registryUrl == null) {
+            registryUrl = binding.getProperties().get("registryurl");
+        }
+        if (registryUrl != null) {
+            properties.put("mp.messaging.connector.smallrye-kafka.apicurio.registry.url", registryUrl);
+        }
+
+        for (String channel : channels) {
+
+            String prefix = channel;
+
+            String oAuthHost = binding.getProperties().get("oauthServerUrl");
+            if (oAuthHost == null) {
+                oAuthHost = binding.getProperties().get("oauthserverurl");
+            }
+            if (oAuthHost != null) {
+                properties.put(prefix + "apicurio.auth.service.url", oAuthHost);
+            }
+
+            String clientId = binding.getProperties().get("clientId");
+            if (clientId == null) {
+                clientId = binding.getProperties().get("clientid");
+            }
+            if (clientId != null) {
+                properties.put(prefix + "apicurio.auth.client.id", clientId);
+            }
+
+            String clientSecret = binding.getProperties().get("clientSecret");
+            if (clientSecret == null) {
+                clientSecret = binding.getProperties().get("clientsecret");
+            }
+            if (clientSecret != null) {
+                properties.put(prefix + "apicurio.auth.client.secret", clientSecret);
+            }
+
+            String realm = binding.getProperties().get("oauthRealm");
+            if (realm == null) {
+                realm = binding.getProperties().get("oauthRealm");
+            }
+            if (clientSecret != null) {
+                properties.put(prefix + "apicurio.auth.realm", realm);
+            }
+
+            if (registryUrl != null) {
+                properties.put(prefix + "apicurio.registry.url", registryUrl);
+            }
+        }
+
+        return Optional.of(new ServiceBindingConfigSource("serviceregistry-k8s-service-binding-source", properties));
+    }
+
+    private List<String> extractChannels(Config configIn) {
+
+        var list = new ArrayList<String>();
+
+        for (String propertyName : configIn.getPropertyNames()) {
```

Comment:
While this is working, I'm wondering if we shouldn't rely on the new build items we have that would provide the list of incoming and outgoing channels.

However, passing these channels to that class (as it's an SPI) can be convoluted.

Reply:
@radcortez does the getPropertyNames returns an empty iterable or `null` if there are no names?

Reply:
An empty iterable.